### PR TITLE
Add consensus state input checks

### PR DESCRIPTION
**Pull request by gabrielbosio**

### CODE REVIEW

File: operator/mina/lib/src/consensus_state.rs [hunk 1]
```diff
@@ -171,4 +190,115 @@ mod test {
...
         );
     }
+
+    #[test]
+    fn test_candidate_state_with_smaller_global_slot_than_tip_state() {
+        let valid_proof: MinaStateProof = bincode::deserialize(PROOF_BYTES).unwrap();
+        let new_tip = valid_proof.bridge_tip_state;
+        let mut old_tip = valid_proof.candidate_chain_states.last().unwrap().clone();
+
+        // Force checking longe fork rule:
```
Comment (xqft) [inline operator/mina/lib/src/consensus_state.rs]:
```suggestion
        // Force checking long fork rule:
```

File: operator/mina/lib/src/consensus_state.rs [hunk 2]
```diff
@@ -171,4 +190,115 @@ mod test {
...
+        );
+    }
+
+    #[test]
+    fn test_candidate_state_with_less_sub_windows_densities_than_sub_windows_per_window() {
+        let valid_proof: MinaStateProof = bincode::deserialize(PROOF_BYTES).unwrap();
+        let mut new_tip = valid_proof.bridge_tip_state;
+        let mut old_tip = valid_proof.candidate_chain_states.last().unwrap().clone();
+
+        // Force checking longe fork rule:
```
Comment (xqft) [inline operator/mina/lib/src/consensus_state.rs]:
```suggestion
        // Force checking long fork rule:
```

File: operator/mina/lib/src/verifier_index.rs [hunk 3]
```diff
@@ -17,7 +17,7 @@ use kimchi::{
 };
 use serde::Deserialize;
 
-const BLOCKCHAIN_VK_JSON: &str = include_str!("mainnet_vk.json");
+const BLOCKCHAIN_VK_JSON: &str = include_str!("devnet_vk.json");
```
Comment (xqft) [inline operator/mina/lib/src/verifier_index.rs]: This is because we're running the testnet node currently, right?

**Review by xqft — APPROVED** [review]

File: operator/mina/lib/src/verifier_index.rs (hunk 3 above)
Comment (gabrielbosio) [reply operator/mina/lib/src/verifier_index.rs]: Yes, but I can revert this change until we parameterize it.

Comment (xqft) [reply operator/mina/lib/src/verifier_index.rs]: its okay !

**Review by xqft — APPROVED** [review]

**Pull request merged**